### PR TITLE
Build `libtiledbsoma` and `tiledbsoma`-py on POSIX

### PR DESCRIPTION
**Pull request by nguyenv**
Context: https://github.com/single-cell-data/TileDB-SOMA/issues/494

Requires changes in https://github.com/single-cell-data/TileDB-SOMA/pull/1043.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,43 +1,81 @@
...
-  sha256: e0fb16e55e01315a46f102739dcd3d1b15f24d636e00a2d737c0dde674b86370
-
-build:
-  number: 0
-  script: python -m pip install . -vv
-  skip: true  # [win]
+  git_url: https://github.com/single-cell-data/TileDB-SOMA.git
+  git_rev: 8a57a05edecc2b2e17675dc59791c2dedc764034
 
 requirements:
```
Comment (jdblischak) [inline recipe/meta.yaml:12]: You can simplify recipe maintenance and the build process by removing these top-level requirements and combining `build.sh` and `build-libtiledbsome.sh`. My PR https://github.com/TileDB-Inc/tiledb-vcf-feedstock/pull/75 implemented this for the libtiledbvcf recipe

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,43 +1,81 @@
...
+    build:
+      binary_relocation: 
+        - lib/python3.10/site-packages/tiledbsoma/libtiledbsoma.*
+        - lib/python3.10/site-packages/tiledbsoma/pytiledbsoma.*.so
+    requirements:
+      build:
+        - {{ compiler('cxx') }}
+        - {{ compiler('c') }}
+        - cmake >=3.21
+        - make
```
Comment (jdblischak) [inline recipe/meta.yaml:56]: Are `cmake` and `make` build dependencies for the Python client when it is building against a pre-built libtiledbsoma? As a related example, building tiledbvcf-py doesn't require `cmake` and `make` when it builds against a pre-built libtiledbvcf (https://github.com/TileDB-Inc/tiledb-vcf-feedstock/pull/78)

**Review by johnkerl — APPROVED** [review]
lgtm except
https://github.com/single-cell-data/TileDB-SOMA/commit/123cb4370057eb47c5657ca2f846336a86e6032e
has not yet been tagged as 1.0.1

**Review by johnkerl — APPROVED** [review]

**Comment by johnkerl** [comment]
For the record the SHA on here is for https://github.com/single-cell-data/TileDB-SOMA/releases/tag/1.1.1

**Pull request merged**